react/features/e2ee/middleware.js

```diff
@@ -35,6 +46,69 @@ MiddlewareRegistry.register(({ dispatch, getState }) => next => action => {
         dispatch(unregisterSound(E2EE_ON_SOUND_ID));
         break;
 
+    case PARTICIPANT_UPDATED:
+    case PARTICIPANT_JOINED: {
+        const result = next(action);
+        const { e2eeEnabled, e2eeSupported, local } = action.participant;
+        const { everyoneEnabledE2EE } = getState()['features/e2ee'];
+
+        // if all had it enabled and this one disabled it, change value in store
+        // otherwise there is no change in the value we store
+        if (everyoneEnabledE2EE && !e2eeEnabled) {
+            dispatch({
+                type: SET_EVERYONE_ENABLED,
+                everyoneEnabledE2EE: false
+            });
+        }
+
+        if (local) {
+            return result;
+        }
+
+        const { everyoneSupportsE2EE } = getState()['features/e2ee'];
+
+        // if all supported it and this one does not, change value in store
+        // otherwise there is no change in the value we store
+        if (everyoneSupportsE2EE && !e2eeSupported) {
+            dispatch({
+                type: SET_EVERYONE_SUPPORTS,
+                everyoneSupportsE2EE: false
+            });
+        }
+
+        return result;
+    }
+
+    case PARTICIPANT_LEFT: {
+        const result = next(action);
+        const { e2eeEnabled, e2eeSupported } = action.participant;
+
+        const { everyoneEnabledE2EE } = getState()['features/e2ee'];
+
+        const participants = getParticipants(getState());
```

Comment:
I'm  trying reduce the usage of `getParticipants` or even completely remove it. The problem is that `getParticipants` returns array and in store we use `Map`. It basically converts the `Map` to a new array which I'm not sure how expensive is. Anyway in most cases we can loop trough the Map directly. Could you please try to use the `Map` directly?

Reply:
Will fix it.